src/App.tsx

```diff
@@ -1,44 +1,97 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/bulma.sass';
 import '@fortawesome/fontawesome-free/css/all.css';
 import './App.scss';
-
 import classNames from 'classnames';
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { User } from './types/User';
+import { client } from './utils/fetchClient';
+import { Post } from './types/Post';
 
 export const App: React.FC = () => {
+  const [selectedUser, setSelectedUser] = useState<null | User>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [users, setUsers] = useState<User[]>([]);
+  const [userPosts, setUserPosts] = useState<Post[]>([]);
+  const [hasError, setHasError] = useState(false);
+  const [isNoPosts, setIsNoPosts] = useState(false);
+  const [isSideBar, setIsSideBar] = useState(false);
+  const [currentPost, setCurrentPost] = useState<Post | null>(null);
+  const [hasCommentForm, setHasCommentForm] = useState(false);
+
+  useEffect(() => {
+    client.get<User[]>('/users')
+      .then(usersItems => {
+        setUsers(usersItems);
```

Comment:
also, you can do it in this way:
```suggestion
      .then(setUsers);
```